python_modules/dagster/dagster/_core/blueprints/shell_command_asset_blueprint.py

```diff
@@ -0,0 +1,61 @@
+from typing import AbstractSet, Literal, Mapping, Optional, Sequence, Union
+
+from dagster import AssetExecutionContext, MaterializeResult
+from dagster._core.blueprints.base_asset_blueprint import (
+    BaseAssetBlueprint,
+    BaseMultiAssetBlueprint,
+)
+
+
+class ShellCommandAssetBlueprint(BaseAssetBlueprint):
+    """A blueprint for an asset definition whose materialization function is a shell command.
+
+    Requires the code location to include a "pipes_subprocess_client" resource with type
+    dagster.PipesSubprocessClient.
+
+    Attributes:
+        command (Union[str, Sequence[str]]): The command to run. Will be passed to `subprocess.Popen()`.
+        env (Optional[Mapping[str, str]]): An optional dict of environment variables to pass to the subprocess.
+        cwd (Optional[str]): Working directory in which to launch the subprocess command.
+    """
+
+    type: Literal["shell_command_asset"] = "shell_command_asset"
```

Comment:
do we want to namespace the `type` here somehow, if we expect users to `Union` a bunch of `BaseAssetBlueprint`s together from base libs + their own code?

Reply:
Good point. Made this change.